kubernetes/structures_stateful_set.go

```diff
@@ -0,0 +1,193 @@
+		obj.ServiceName = v
+	}
+
+	if v, ok := in["update_strategy"].(map[string]interface{}); ok {
+		ust := v1.StatefulSetUpdateStrategy{}
+		ust.Type = v["type"].(v1.StatefulSetUpdateStrategyType)
+		if r, ok := v["rolling_update"].(map[string]interface{}); ok {
+			s := v1.RollingUpdateStatefulSetStrategy{}
+			if p, ok := r["partition"].(int32); ok {
+				s.Partition = ptrToInt32(int32(p))
+			}
+			ust.RollingUpdate = &s
+		}
+		obj.UpdateStrategy = ust
+	}
+
+	template, err := expandPodTemplate(in["template"].([]interface{}))
+	if err != nil {
+		return obj, err
+	}
+	obj.Template = template
+
+	if v, ok := in["volume_claim_template"].([]interface{}); ok {
+		obj.VolumeClaimTemplates = []corev1.PersistentVolumeClaim{}
+		if len(v) == 0 || v[0] == nil {
+			return obj, nil
+		}
+		for _, pvc := range v {
+			p, err := expandPersistenVolumeClaim(pvc.([]interface{}))
+			if err != nil {
+				return obj, err
+			}
+			obj.VolumeClaimTemplates = append(obj.VolumeClaimTemplates, p)
+		}
+	}
+	return obj, nil
+}
+
+func expandPersistenVolumeClaim(p []interface{}) (corev1.PersistentVolumeClaim, error) {
+	if len(p) == 0 || p[0] == nil {
+		return corev1.PersistentVolumeClaim{}, nil
+	}
+	in := p[0].(map[string]interface{})
+	metadata := expandMetadata(in["metadata"].([]interface{}))
+	spec, err := expandPersistentVolumeClaimSpec(in["spec"].([]interface{}))
+	if err != nil {
+		return corev1.PersistentVolumeClaim{}, err
+	}
+	pvc := corev1.PersistentVolumeClaim{
+		ObjectMeta: metadata,
+		Spec:       spec,
+	}
+	return pvc, nil
+}
+
+func expandStatefulSetSelectors(s []interface{}) (metav1.LabelSelector, error) {
+	obj := metav1.LabelSelector{}
+	if len(s) == 0 || s[0] == nil {
+		return obj, nil
+	}
+	in := s[0].(map[string]interface{})
+	log.Printf("[DEBUG] StatefulSet Selector: %#v", in)
+	if v, ok := in["match_labels"].(map[string]interface{}); ok {
+		log.Printf("[DEBUG] StatefulSet Selector MatchLabels: %#v", v)
+		ml := make(map[string]string)
+		for k, l := range v {
+			ml[k] = l.(string)
+			log.Printf("[DEBUG] StatefulSet Selector MatchLabel: %#v -> %#v", k, v)
+		}
+		obj.MatchLabels = ml
+	}
+	if v, ok := in["match_expressions"].([]interface{}); ok {
+		log.Printf("[DEBUG] StatefulSet Selector MatchExpressions: %#v", v)
+		me, err := expandMatchExpressions(v)
+		if err != nil {
+			return obj, err
+		}
+		obj.MatchExpressions = me
+	}
+	return obj, nil
+}
+
+func expandMatchExpressions(in []interface{}) ([]metav1.LabelSelectorRequirement, error) {
+	if len(in) == 0 {
+		return []metav1.LabelSelectorRequirement{}, nil
+	}
+	obj := make([]metav1.LabelSelectorRequirement, len(in))
+	for i, c := range in {
+		p := c.(map[string]interface{})
+		if v, ok := p["key"].(string); ok {
+			obj[i].Key = v
+		}
+		if v, ok := p["operator"].(metav1.LabelSelectorOperator); ok {
+			obj[i].Operator = v
+		}
+		if v, ok := p["values"].(*schema.Set); ok {
+			obj[i].Values = schemaSetToStringArray(v)
+		}
+	}
+	return obj, nil
+}
+
+// Flattners
```

Comment:
I think this comment may confuse godoc

Reply:
We seem to have this convention throughout the provider.
I think `godoc` requires that the comment and the affected code block don't have any whitespace in-between.